### PR TITLE
Use IReadOnlyList instead of ImmutableArray

### PR DESCRIPTION
**Pull request by davkean**
(Preparing to make IActiveConfiguredProjectsProvider public for the VSTest guys)

CPS doesn't expose ImmutableArray in their API. Move to IReadOnlyList<T> to be consistent with them.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectConfigurationFactory.cs [hunk 1]
```diff
@@ -28,9 +29,9 @@ public static ProjectConfiguration Create(string configuration)
             return Create(configuration, dimensionsBuilder.ToImmutable());
         }
 
-        public static IImmutableSet<ProjectConfiguration> CreateMany(params string[] configurations)
+        public static IReadOnlyList<ProjectConfiguration> CreateMany(params string[] configurations)
         {
-            var configurationsBuilder = ImmutableHashSet.CreateBuilder<ProjectConfiguration>();
+            var configurationsBuilder = ImmutableArray.CreateBuilder<ProjectConfiguration>();
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectConfigurationFactory.cs]: In theory the hash set would have enforced that the same `ProjectConfiguration` wasn't added twice, assuming the underlying implementation of the `ProjectConfiguration` interface actually implemented a proper `GetHashCode`.

Does this need to be enforced in some other way within `CreateMany` or do we just say that the caller needs to ensure there are no duplicates in the string array it passes in?

Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectConfigurationFactory.cs]: This change is not for public API. This is just for the Mock. So this change is ok and I guess, we dont need to do anything.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectConfigurationFactory.cs]: Just a mock change.

**Pull request merged**